app/full_text_search.py

```diff
@@ -0,0 +1,61 @@
+from const import PERSON_FULLTEXT_INDEX_NAME
+
+from google.appengine.api import search
+import re
+import model
+
+def search_with_index(repo, query_obj, max_results):
+
+    def create_query(query):
+        return re.sub(r' ', ' AND ', query)
+
+    results = []
+    query = create_query(query_obj)
+    try:
+        index = search.Index(name=PERSON_FULLTEXT_INDEX_NAME)
+        options = search.QueryOptions(
+            limit=max_results,
+            returned_fields=['record_id'])
+        results_index = index.search(search.Query(
+            query_string=query, options=options))
+        record_ids = []
+        for document in results_index:
+            record_ids.append(document.fields[0].value)
+        for id in record_ids:
+            results.append(model.Person.get_by_key_name(repo + ':' + id))
+    except search.Error:
+        logging.exception('Search exception')
+    return results
+
+
+def create_document(**kwargs):
+    return search.Document(
```

Comment:
FIELDS_TO_INDEX = ['record_id', ...]

for field in FIELDS_TO_INDEX:
  if field in kwargs:
    fields.append(search.TextField(name=field, value=kwargs[field])

Or

for key, value in kwargs:
  if key in FIELDS_TO_INDEX:
    fields.append(search.TextField(name=key, value=value)